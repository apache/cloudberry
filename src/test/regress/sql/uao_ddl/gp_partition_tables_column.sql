-- test legacy/gp-style partition tables

create schema gppt_ao_column;
set search_path="$user",gppt_ao_column,public;

-- set default table access method to non-ao/co table
SET default_table_access_method=heap;

CREATE TABLE gppt_ao_column.pt_ao_column (
a date NOT NULL,
b integer NOT NULL,
c numeric(20,10),
d integer,
bi bigint NOT NULL,
ts timestamp DEFAULT CURRENT_TIMESTAMP NOT NULL
)
using ao_column
WITH (compresstype=zlib, compresslevel=6)
DISTRIBUTED RANDOMLY
PARTITION BY RANGE (a) (
START ('2020-01-01'::date) INCLUSIVE
END ('2024-01-01'::date) EXCLUSIVE
EVERY (interval '1 Year')
)
;

select relname, relkind, amname, reloptions from pg_class c left join pg_am am on c.relam=am.oid where relname='pt_ao_column';
\d+ gppt_ao_column.pt_ao_column
\d+ gppt_ao_column.pt_ao_column_1_prt_2

ALTER TABLE gppt_ao_column.pt_ao_column ADD PARTITION START ('2027-01-01') INCLUSIVE END ('2028-01-01') EXCLUSIVE;
\d+ gppt_ao_column.pt_ao_column_1_prt_11
select relname, relkind, amname, reloptions from pg_class c left join pg_am am on c.relam=am.oid where relname='pt_ao_column_1_prt_11';

CREATE TABLE gppt_ao_column.pt2_ao_column (
a date NOT NULL,
b integer NOT NULL,
c numeric(20,10),
d integer,
bi bigint NOT NULL,
ts timestamp DEFAULT CURRENT_TIMESTAMP NOT NULL
)
using ao_column
DISTRIBUTED RANDOMLY
PARTITION BY RANGE (a) (
START ('2020-01-01'::date) INCLUSIVE
END ('2024-01-01'::date) EXCLUSIVE
EVERY (interval '1 Year')
)
;

select relname, relkind, amname, reloptions from pg_class c left join pg_am am on c.relam=am.oid where relname='pt2_ao_column';
\d+ gppt_ao_column.pt2_ao_column
\d+ gppt_ao_column.pt2_ao_column_1_prt_2

ALTER TABLE gppt_ao_column.pt2_ao_column ADD PARTITION START ('2027-01-01') INCLUSIVE END ('2028-01-01') EXCLUSIVE;
\d+ gppt_ao_column.pt2_ao_column
\d+ gppt_ao_column.pt2_ao_column_1_prt_11
select relname, relkind, amname, reloptions from pg_class c left join pg_am am on c.relam=am.oid where relname='pt2_ao_column_1_prt_11';

create table gppt_ao_column.alter_part_tab1 (id SERIAL, a1 int, a2 char(5), a3 text)
  WITH (appendonly=true, orientation=column, compresstype=zlib)
  partition by list(a2) subpartition by range(a1)
  subpartition template (
    default subpartition subothers,
    subpartition sp1 start(1) end(9) with(appendonly=true,orientation=column,compresstype=rle_type),
    subpartition sp2 start(11) end(20) with(appendonly=true,orientation=column,compresstype=zstd))
   (partition p1 values('val1') , partition p2 values('val2'));

create index on gppt_ao_column.alter_part_tab1(a1);
create index on gppt_ao_column.alter_part_tab1 using bitmap(a3);
alter table gppt_ao_column.alter_part_tab1 add column a4 numeric default 5.5;
update gppt_ao_column.alter_part_tab1 set a4 = a1 % 2;
ALTER TABLE gppt_ao_column.alter_part_tab1 ADD partition p31 values(1);
\d+ gppt_ao_column.alter_part_tab1
\d+ gppt_ao_column.alter_part_tab1_1_prt_p31


reset default_table_access_method;
reset search_path;
