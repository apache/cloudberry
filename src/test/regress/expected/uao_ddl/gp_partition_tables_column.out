-- test legacy/gp-style partition tables
create schema gppt_ao_column;
set search_path="$user",gppt_ao_column,public;
-- set default table access method to non-ao/co table
SET default_table_access_method=heap;
CREATE TABLE gppt_ao_column.pt_ao_column (
a date NOT NULL,
b integer NOT NULL,
c numeric(20,10),
d integer,
bi bigint NOT NULL,
ts timestamp DEFAULT CURRENT_TIMESTAMP NOT NULL
)
using ao_column
WITH (compresstype=zlib, compresslevel=6)
DISTRIBUTED RANDOMLY
PARTITION BY RANGE (a) (
START ('2020-01-01'::date) INCLUSIVE
END ('2024-01-01'::date) EXCLUSIVE
EVERY (interval '1 Year')
)
;
select relname, relkind, amname, reloptions from pg_class c left join pg_am am on c.relam=am.oid where relname='pt_ao_column';
   relname    | relkind |  amname   |             reloptions              
--------------+---------+-----------+-------------------------------------
 pt_ao_column | p       | ao_column | {compresstype=zlib,compresslevel=6}
(1 row)

\d+ gppt_ao_column.pt_ao_column
                                                              Partitioned table "gppt_ao_column.pt_ao_column"
 Column |            Type             | Collation | Nullable |      Default      | Storage | Stats target | Compression Type | Compression Level | Block Size | Description 
--------+-----------------------------+-----------+----------+-------------------+---------+--------------+------------------+-------------------+------------+-------------
 a      | date                        |           | not null |                   | plain   |              | zlib             | 6                 | 32768      | 
 b      | integer                     |           | not null |                   | plain   |              | zlib             | 6                 | 32768      | 
 c      | numeric(20,10)              |           |          |                   | main    |              | zlib             | 6                 | 32768      | 
 d      | integer                     |           |          |                   | plain   |              | zlib             | 6                 | 32768      | 
 bi     | bigint                      |           | not null |                   | plain   |              | zlib             | 6                 | 32768      | 
 ts     | timestamp without time zone |           | not null | CURRENT_TIMESTAMP | plain   |              | zlib             | 6                 | 32768      | 
Partition key: RANGE (a)
Partitions: pt_ao_column_1_prt_1 FOR VALUES FROM ('01-01-2020') TO ('01-01-2021'),
            pt_ao_column_1_prt_2 FOR VALUES FROM ('01-01-2021') TO ('01-01-2022'),
            pt_ao_column_1_prt_3 FOR VALUES FROM ('01-01-2022') TO ('01-01-2023'),
            pt_ao_column_1_prt_4 FOR VALUES FROM ('01-01-2023') TO ('01-01-2024')
Distributed randomly
Options: compresstype=zlib, compresslevel=6

\d+ gppt_ao_column.pt_ao_column_1_prt_2
                                                                Table "gppt_ao_column.pt_ao_column_1_prt_2"
 Column |            Type             | Collation | Nullable |      Default      | Storage | Stats target | Compression Type | Compression Level | Block Size | Description 
--------+-----------------------------+-----------+----------+-------------------+---------+--------------+------------------+-------------------+------------+-------------
 a      | date                        |           | not null |                   | plain   |              | zlib             | 6                 | 32768      | 
 b      | integer                     |           | not null |                   | plain   |              | zlib             | 6                 | 32768      | 
 c      | numeric(20,10)              |           |          |                   | main    |              | zlib             | 6                 | 32768      | 
 d      | integer                     |           |          |                   | plain   |              | zlib             | 6                 | 32768      | 
 bi     | bigint                      |           | not null |                   | plain   |              | zlib             | 6                 | 32768      | 
 ts     | timestamp without time zone |           | not null | CURRENT_TIMESTAMP | plain   |              | zlib             | 6                 | 32768      | 
Partition of: pt_ao_column FOR VALUES FROM ('01-01-2021') TO ('01-01-2022')
Partition constraint: ((a IS NOT NULL) AND (a >= '01-01-2021'::date) AND (a < '01-01-2022'::date))
Checksum: t
Distributed randomly
Options: compresstype=zlib, compresslevel=6

ALTER TABLE gppt_ao_column.pt_ao_column ADD PARTITION START ('2027-01-01') INCLUSIVE END ('2028-01-01') EXCLUSIVE;
\d+ gppt_ao_column.pt_ao_column_1_prt_11
                                                                Table "gppt_ao_column.pt_ao_column_1_prt_11"
 Column |            Type             | Collation | Nullable |      Default      | Storage | Stats target | Compression Type | Compression Level | Block Size | Description 
--------+-----------------------------+-----------+----------+-------------------+---------+--------------+------------------+-------------------+------------+-------------
 a      | date                        |           | not null |                   | plain   |              | zlib             | 6                 | 32768      | 
 b      | integer                     |           | not null |                   | plain   |              | zlib             | 6                 | 32768      | 
 c      | numeric(20,10)              |           |          |                   | main    |              | zlib             | 6                 | 32768      | 
 d      | integer                     |           |          |                   | plain   |              | zlib             | 6                 | 32768      | 
 bi     | bigint                      |           | not null |                   | plain   |              | zlib             | 6                 | 32768      | 
 ts     | timestamp without time zone |           | not null | CURRENT_TIMESTAMP | plain   |              | zlib             | 6                 | 32768      | 
Partition of: pt_ao_column FOR VALUES FROM ('01-01-2027') TO ('01-01-2028')
Partition constraint: ((a IS NOT NULL) AND (a >= '01-01-2027'::date) AND (a < '01-01-2028'::date))
Checksum: t
Distributed randomly
Options: compresstype=zlib, compresslevel=6

select relname, relkind, amname, reloptions from pg_class c left join pg_am am on c.relam=am.oid where relname='pt_ao_column_1_prt_11';
        relname        | relkind |  amname   |             reloptions              
-----------------------+---------+-----------+-------------------------------------
 pt_ao_column_1_prt_11 | r       | ao_column | {compresstype=zlib,compresslevel=6}
(1 row)

CREATE TABLE gppt_ao_column.pt2_ao_column (
a date NOT NULL,
b integer NOT NULL,
c numeric(20,10),
d integer,
bi bigint NOT NULL,
ts timestamp DEFAULT CURRENT_TIMESTAMP NOT NULL
)
using ao_column
DISTRIBUTED RANDOMLY
PARTITION BY RANGE (a) (
START ('2020-01-01'::date) INCLUSIVE
END ('2024-01-01'::date) EXCLUSIVE
EVERY (interval '1 Year')
)
;
select relname, relkind, amname, reloptions from pg_class c left join pg_am am on c.relam=am.oid where relname='pt2_ao_column';
    relname    | relkind |  amname   | reloptions 
---------------+---------+-----------+------------
 pt2_ao_column | p       | ao_column | 
(1 row)

\d+ gppt_ao_column.pt2_ao_column
                                                              Partitioned table "gppt_ao_column.pt2_ao_column"
 Column |            Type             | Collation | Nullable |      Default      | Storage | Stats target | Compression Type | Compression Level | Block Size | Description 
--------+-----------------------------+-----------+----------+-------------------+---------+--------------+------------------+-------------------+------------+-------------
 a      | date                        |           | not null |                   | plain   |              | none             | 0                 | 32768      | 
 b      | integer                     |           | not null |                   | plain   |              | none             | 0                 | 32768      | 
 c      | numeric(20,10)              |           |          |                   | main    |              | none             | 0                 | 32768      | 
 d      | integer                     |           |          |                   | plain   |              | none             | 0                 | 32768      | 
 bi     | bigint                      |           | not null |                   | plain   |              | none             | 0                 | 32768      | 
 ts     | timestamp without time zone |           | not null | CURRENT_TIMESTAMP | plain   |              | none             | 0                 | 32768      | 
Partition key: RANGE (a)
Partitions: pt2_ao_column_1_prt_1 FOR VALUES FROM ('01-01-2020') TO ('01-01-2021'),
            pt2_ao_column_1_prt_2 FOR VALUES FROM ('01-01-2021') TO ('01-01-2022'),
            pt2_ao_column_1_prt_3 FOR VALUES FROM ('01-01-2022') TO ('01-01-2023'),
            pt2_ao_column_1_prt_4 FOR VALUES FROM ('01-01-2023') TO ('01-01-2024')
Distributed randomly

\d+ gppt_ao_column.pt2_ao_column_1_prt_2
                                                                Table "gppt_ao_column.pt2_ao_column_1_prt_2"
 Column |            Type             | Collation | Nullable |      Default      | Storage | Stats target | Compression Type | Compression Level | Block Size | Description 
--------+-----------------------------+-----------+----------+-------------------+---------+--------------+------------------+-------------------+------------+-------------
 a      | date                        |           | not null |                   | plain   |              | none             | 0                 | 32768      | 
 b      | integer                     |           | not null |                   | plain   |              | none             | 0                 | 32768      | 
 c      | numeric(20,10)              |           |          |                   | main    |              | none             | 0                 | 32768      | 
 d      | integer                     |           |          |                   | plain   |              | none             | 0                 | 32768      | 
 bi     | bigint                      |           | not null |                   | plain   |              | none             | 0                 | 32768      | 
 ts     | timestamp without time zone |           | not null | CURRENT_TIMESTAMP | plain   |              | none             | 0                 | 32768      | 
Partition of: pt2_ao_column FOR VALUES FROM ('01-01-2021') TO ('01-01-2022')
Partition constraint: ((a IS NOT NULL) AND (a >= '01-01-2021'::date) AND (a < '01-01-2022'::date))
Checksum: t
Distributed randomly

ALTER TABLE gppt_ao_column.pt2_ao_column ADD PARTITION START ('2027-01-01') INCLUSIVE END ('2028-01-01') EXCLUSIVE;
\d+ gppt_ao_column.pt2_ao_column
                                                              Partitioned table "gppt_ao_column.pt2_ao_column"
 Column |            Type             | Collation | Nullable |      Default      | Storage | Stats target | Compression Type | Compression Level | Block Size | Description 
--------+-----------------------------+-----------+----------+-------------------+---------+--------------+------------------+-------------------+------------+-------------
 a      | date                        |           | not null |                   | plain   |              | none             | 0                 | 32768      | 
 b      | integer                     |           | not null |                   | plain   |              | none             | 0                 | 32768      | 
 c      | numeric(20,10)              |           |          |                   | main    |              | none             | 0                 | 32768      | 
 d      | integer                     |           |          |                   | plain   |              | none             | 0                 | 32768      | 
 bi     | bigint                      |           | not null |                   | plain   |              | none             | 0                 | 32768      | 
 ts     | timestamp without time zone |           | not null | CURRENT_TIMESTAMP | plain   |              | none             | 0                 | 32768      | 
Partition key: RANGE (a)
Partitions: pt2_ao_column_1_prt_1 FOR VALUES FROM ('01-01-2020') TO ('01-01-2021'),
            pt2_ao_column_1_prt_11 FOR VALUES FROM ('01-01-2027') TO ('01-01-2028'),
            pt2_ao_column_1_prt_2 FOR VALUES FROM ('01-01-2021') TO ('01-01-2022'),
            pt2_ao_column_1_prt_3 FOR VALUES FROM ('01-01-2022') TO ('01-01-2023'),
            pt2_ao_column_1_prt_4 FOR VALUES FROM ('01-01-2023') TO ('01-01-2024')
Distributed randomly

\d+ gppt_ao_column.pt2_ao_column_1_prt_11
                                                               Table "gppt_ao_column.pt2_ao_column_1_prt_11"
 Column |            Type             | Collation | Nullable |      Default      | Storage | Stats target | Compression Type | Compression Level | Block Size | Description 
--------+-----------------------------+-----------+----------+-------------------+---------+--------------+------------------+-------------------+------------+-------------
 a      | date                        |           | not null |                   | plain   |              | none             | 0                 | 32768      | 
 b      | integer                     |           | not null |                   | plain   |              | none             | 0                 | 32768      | 
 c      | numeric(20,10)              |           |          |                   | main    |              | none             | 0                 | 32768      | 
 d      | integer                     |           |          |                   | plain   |              | none             | 0                 | 32768      | 
 bi     | bigint                      |           | not null |                   | plain   |              | none             | 0                 | 32768      | 
 ts     | timestamp without time zone |           | not null | CURRENT_TIMESTAMP | plain   |              | none             | 0                 | 32768      | 
Partition of: pt2_ao_column FOR VALUES FROM ('01-01-2027') TO ('01-01-2028')
Partition constraint: ((a IS NOT NULL) AND (a >= '01-01-2027'::date) AND (a < '01-01-2028'::date))
Checksum: t
Distributed randomly

select relname, relkind, amname, reloptions from pg_class c left join pg_am am on c.relam=am.oid where relname='pt2_ao_column_1_prt_11';
        relname         | relkind |  amname   | reloptions 
------------------------+---------+-----------+------------
 pt2_ao_column_1_prt_11 | r       | ao_column | 
(1 row)

create table gppt_ao_column.alter_part_tab1 (id SERIAL, a1 int, a2 char(5), a3 text)
  WITH (appendonly=true, orientation=column, compresstype=zlib)
  partition by list(a2) subpartition by range(a1)
  subpartition template (
    default subpartition subothers,
    subpartition sp1 start(1) end(9) with(appendonly=true,orientation=column,compresstype=rle_type),
    subpartition sp2 start(11) end(20) with(appendonly=true,orientation=column,compresstype=zstd))
   (partition p1 values('val1') , partition p2 values('val2'));
create index on gppt_ao_column.alter_part_tab1(a1);
create index on gppt_ao_column.alter_part_tab1 using bitmap(a3);
alter table gppt_ao_column.alter_part_tab1 add column a4 numeric default 5.5;
update gppt_ao_column.alter_part_tab1 set a4 = a1 % 2;
ALTER TABLE gppt_ao_column.alter_part_tab1 ADD partition p31 values(1);
\d+ gppt_ao_column.alter_part_tab1
                                                                   Partitioned table "gppt_ao_column.alter_part_tab1"
 Column |     Type     | Collation | Nullable |                   Default                   | Storage  | Stats target | Compression Type | Compression Level | Block Size | Description 
--------+--------------+-----------+----------+---------------------------------------------+----------+--------------+------------------+-------------------+------------+-------------
 id     | integer      |           | not null | nextval('alter_part_tab1_id_seq'::regclass) | plain    |              | zlib             | 1                 | 32768      | 
 a1     | integer      |           |          |                                             | plain    |              | zlib             | 1                 | 32768      | 
 a2     | character(5) |           |          |                                             | extended |              | zlib             | 1                 | 32768      | 
 a3     | text         |           |          |                                             | extended |              | zlib             | 1                 | 32768      | 
 a4     | numeric      |           |          | 5.5                                         | main     |              | none             | 0                 | 32768      | 
Partition key: LIST (a2)
Indexes:
    "alter_part_tab1_a1_idx" btree (a1)
    "alter_part_tab1_a3_idx" bitmap (a3)
Partitions: alter_part_tab1_1_prt_p1 FOR VALUES IN ('val1 '), PARTITIONED,
            alter_part_tab1_1_prt_p2 FOR VALUES IN ('val2 '), PARTITIONED,
            alter_part_tab1_1_prt_p31 FOR VALUES IN ('1    '), PARTITIONED
Distributed by: (id)
Options: compresstype=zlib

\d+ gppt_ao_column.alter_part_tab1_1_prt_p31
                                                              Partitioned table "gppt_ao_column.alter_part_tab1_1_prt_p31"
 Column |     Type     | Collation | Nullable |                   Default                   | Storage  | Stats target | Compression Type | Compression Level | Block Size | Description 
--------+--------------+-----------+----------+---------------------------------------------+----------+--------------+------------------+-------------------+------------+-------------
 id     | integer      |           | not null | nextval('alter_part_tab1_id_seq'::regclass) | plain    |              | zlib             | 1                 | 32768      | 
 a1     | integer      |           |          |                                             | plain    |              | zlib             | 1                 | 32768      | 
 a2     | character(5) |           |          |                                             | extended |              | zlib             | 1                 | 32768      | 
 a3     | text         |           |          |                                             | extended |              | zlib             | 1                 | 32768      | 
 a4     | numeric      |           |          | 5.5                                         | main     |              | none             | 0                 | 32768      | 
Partition of: alter_part_tab1 FOR VALUES IN ('1    ')
Partition constraint: ((a2 IS NOT NULL) AND (a2 = '1    '::character(5)))
Partition key: RANGE (a1)
Indexes:
    "alter_part_tab1_1_prt_p31_a1_idx" btree (a1)
    "alter_part_tab1_1_prt_p31_a3_idx" bitmap (a3)
Partitions: alter_part_tab1_1_prt_p31_2_prt_sp1 FOR VALUES FROM (1) TO (9),
            alter_part_tab1_1_prt_p31_2_prt_sp2 FOR VALUES FROM (11) TO (20),
            alter_part_tab1_1_prt_p31_2_prt_subothers DEFAULT
Distributed by: (id)
Options: compresstype=zlib

reset default_table_access_method;
reset search_path;
